test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js

```diff
@@ -310,4 +324,46 @@ test
       record => record.response.statusCode === 200 &&
         record.request.url.match(/poll/)
     )).eql(5);
+  });
+
+test
+  .requestHooks(logger, tooManyRequestMock)('pause polling when encounter 429 too many request', async t => {
+    await setup(t);
+
+    // Encounter 429
+    await t.expect(logger.count(
+      record => record.response.statusCode === 429 &&
+        record.request.url.match(/poll/)
+    )).eql(1);
+
+    await t.removeRequestHooks(tooManyRequestMock);
+    await t.addRequestHooks(validOTPmock);
+
+    // Pause for 60 sec before sending request
+    await t.wait(60000);
```

Comment:
Is this a real 60 seconds, or is the clock being mocked? Could these tests be done in jest?

Reply:
real 60 secs...i thought it might be okay since we have above with 30 seconds wait as well. I am open to move it if possible.